File: dt-spark-core/src/main/scala/com/kevin/scala/transformations/ParallelizePairs_RDD.scala
package com.kevin.scala.transformations

import org.apache.spark.{SparkConf, SparkContext}

/**
  * java的parallelizePairs在scala中没有，所以还是parallelize，
  *     parallelize将Tuple2<K,V>类型的list转为成RDD
  */
object ParallelizePairs_RDD {

  def main(args: Array[String]): Unit = {

    // 1.创建SparkConf
    val conf = new SparkConf().setAppName("ParallelizePairs_RDD").setMaster("local")
    // 2.创建SparkContext
    val sc = new SparkContext(conf)
    // 3.创建tuple2元组集合
    val list = List((0, "aa"),(1, "a"),(2, "b"),(3, "c"))
    // 4.parallelize将Tuple2类型的list转为成RDD,foreach遍历输出
    sc.parallelize(list).foreach(value => (println(value._1+","+value._2)))
    // 5.关闭sc
    sc.stop()

  }

}

File: dt-spark-core/src/main/java/com/kevin/java/transformations/ParallelizePairs_RDD.java
package com.kevin.java.transformations;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.VoidFunction;
import scala.Tuple2;

import java.util.Arrays;

/**
 * @author kevin
 * @version 1.0
 * @description     ParallelizePairs将Tuple2<K,V>类型的list转为成RDD
 * @createDate 2018/12/28
 */
public class ParallelizePairs_RDD {

    public static void main(String[] args) {

        // 1.创建SparkConf，设置作业名称和模式
        SparkConf conf = new SparkConf().setAppName("ParallelizePairs_RDD").setMaster("local");

        // 2.基于Sparkconf对象创建一个SparkContext上下文，它是通往集群的唯一通道，且在创建时会创建任务调度器
        JavaSparkContext sc = new JavaSparkContext(conf);

        // 3.将Tuple2类型的list转为成RDD
        JavaPairRDD<Integer, String> parallelizePairs = sc.parallelizePairs(Arrays.asList(
                new Tuple2<Integer, String>(0, "aa"),
                new Tuple2<Integer, String>(1, "a"),
                new Tuple2<Integer, String>(2, "b"),
                new Tuple2<Integer, String>(3, "c")));

        // 4.遍历数据
        parallelizePairs.foreach(new VoidFunction<Tuple2<Integer, String>>() {
            @Override
            public void call(Tuple2<Integer, String> s) throws Exception {
                System.out.println(s);
            }
        });

        // 5.关闭
        sc.close();
    }
}
